### PR TITLE
Implement prettier-tslint

### PR DESCRIPTION
**Pull request by orblazer**
Hello,
I have implement `prettier-tslint`. This read and format correctly typescript (from tslint file)

Thanks.

### CODE REVIEW

**Comment by CiGit** [comment]
Hi, Thanks for doing that.

We would have to keep in mind that `typescript` files are processed by eslint (eslint config files) currently.

**Comment by azz** [comment]
Nice work - let's get `prettier-tslint` shipped!

**Comment by orblazer** [comment]
Ok, this PR is ready to merge ! Thanks @azz for the merge.

**Comment by AlmaasAre** [comment]
ah, been waiting for this! 😻

**Comment by CiGit** [comment]
Could you also update:
- Readme
- Changlog

Add a test. You can duplicate `testEslint` and start from there. 

This implementation makes tslint a higher priority than eslint. Having both enabled would break previous behavior where eslint was _trying_ to read typescript files (assuming it had the right parser).

**Comment by marvinroger** [comment]
Any updates on this?

**Comment by orblazer** [comment]
no i dont have times this days for make that, if an people can make the test, i am so happy :)

**Comment by Faultless** [comment]
@orblazer made a small PR [here](https://github.com/orblazer/prettier-vscode/pull/1)
Hope it helps speed things up!

**Comment by Faultless** [comment]
[This PR](https://github.com/orblazer/prettier-vscode/pull/2) fixes the tslint test. Its working fine with `tslint.json`, which is the standard name for such a file, but it doesn't seem to work with `.yml` files. Maybe this is not part of this PR, I'll have to take a look to make sure when I have time.

**Comment by Faultless** [comment]
Nevermind actually, this is a limitation of [`prettier-tslint`](https://github.com/azz/prettier-tslint#configuration). It should be PR'd as a feature there.

**Comment by Faultless** [comment]
No idea why this test is failing, its working fine on local. Perhaps there's some missing prettier/tslint config on CI that I am unaware of? I'm open to suggestions to resolve this and get it merged asap.

**Comment by Faultless** [comment]
Friendly nudge: can anyone reproduce tests on a linux/osx environment? and/or help with this obscure CI test failure. On windows the error is not reproduceable.

**Comment by ColCh** [comment]
I'm able to reproduce it on Mac OS

**Comment by ColCh** [comment]
@Faultless seems to br working now 😃 

travis job for my commit: https://travis-ci.org/ColCh/prettier-vscode/jobs/385907251

commit itself https://github.com/ColCh/prettier-vscode/commit/3f1f3feb5e4be37558c27985e350778662e1c8b1

waiting for @orblazer to review PR https://github.com/orblazer/prettier-vscode/pull/5

**Comment by Faultless** [comment]
Fixed a small nit [here](https://github.com/orblazer/prettier-vscode/pull/6), but apart from that there should be no reason not to merge this asap

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ All notable changes to the "prettier-vscode" extension will be documented in thi
 
 ## [Unreleased]
 
+## [1.x.x]
```
Comment (CiGit) [inline CHANGELOG.md]: Put it under [Unreleased]

Comment (orblazer) [reply CHANGELOG.md]: Its done :)

**Comment by Faultless** [comment]
@CiGit is this good to go for you? or is there something blocking still?

**Comment by CiGit** [comment]
@azz deserves to review this one :-)

**Comment by leocaseiro** [comment]
it seems "This branch has conflicts that must be resolved"

**Comment by sunnylqm** [comment]
Can we merge this first since @azz had approved this before and what came after were just test code. So if the idea works, the test works, why not let any possible potential issues to be tested by users? @CiGit

**Comment by chenyang-biu** [comment]
Ah ah ah ah cool

**Comment by bohdanbirdie** [comment]
Coming here every day to check if PR was merged :D
@CiGit @azz sorry for bothering, but merge will be highly appreciated!

**Comment by HaoyangFan** [comment]
hi, just want to know if the `prettier tslint` is already integrated into the prettier so that we don't need to install it every time via the npm and just need to turn it on in user's setting, like what we do for the `prettier eslint`?

**Comment by tommck** [comment]
*crickets*

**Comment by cjbassi** [comment]
So I ran into an issue when testing this out where import statements would get mangled when the file was formatted with the given tslint.json:
>{ "rules": { "quotemark": [true, "single"], "ordered-imports": true } }

I was able to recreate the issue with a simple .ts file that has the following lines:
>import Buttons from '../components/Buttons';
>import { newText } from '../actions/actions';

Formatting the file swaps the lines as intended but it also mangles the quotes.

**Comment by chiefGui** [comment]
:(

**Comment by CiGit** [comment]
Thanks for your contribution.

@cjbassi AFAIK Errors should be corrected in dependencies.

**Comment by chiefGui** [comment]
Hey @CiGit 

Thank you for merging this. Do you have an estimate on when you'll be rolling out the new version containing this update?

Thanks.

**Comment by CiGit** [comment]
@chiefGui Faster than it took to merge it :-)
(under 24h if everything goes as I want)

**Comment by sveyret** [comment]
Hi @orblazer 
I tried to use tslint-prettier in vscode, but it looks like the max-line-length tslint rule is not taken into account for formatting line length, and neither is the print-width rule of prettier. Is this something I am doing wrong ?

**Comment by james2406** [comment]
I am experiencing the same issue as @sveyret. With the following tslint.json setup:

```
{
  "extends": ["tslint:recommended"],
  "rules": {
    "max-line-length": [true, { "limit": 140 }]
  }
}
```

**Comment by Faultless** [comment]
> I tried to use tslint-prettier in vscode, but it looks like the max-line-length tslint rule is not taken into account for formatting line length, and neither is the print-width rule of prettier. Is this something I am doing wrong ?

This should probably go into its own issue for tracking/visibility since this PR has already been merged. Just my 2 cents.

**Comment by aNickzz** [comment]
It seems that prettier is ignoring tslint "Indent" setting. Was that supposed to be fixed in ths PR?

**Comment by aNickzz** [comment]
If `prettier.tslintIntegration` is `true,` prettier ignores both its own tab settings (`prettier.useTabs` & `prettier.tabWidth`) _and_ tslint's `indent`.

**Comment by sveyret** [comment]
I haven't experienced (yet) the problem raised by @aNickzz, but created 2 issues for max-line-length and also a problem of semicolon:
https://github.com/prettier/prettier-vscode/issues/557
https://github.com/prettier/prettier-vscode/issues/558

**Comment by azz** [comment]
@sveyret could you move those issues to the [`prettier-tslint`](https://github.com/azz/prettier-tslint) repo?

**Comment by sveyret** [comment]
@azz Done! :smile: 
https://github.com/azz/prettier-tslint/issues/15
https://github.com/azz/prettier-tslint/issues/16

**Pull request merged**